packages/client/src/runtime/core/types/Utils.ts

```diff
@@ -118,6 +118,19 @@ export type PayloadToResult<P, O extends Record<any, any> = RenameAndNestPayload
 
 export type Select<T, U> = T extends U ? T : never
 
+export type Equals<A, B> = (<T>() => T extends A ? 1 : 2) extends <T>() => T extends B ? 1 : 2 ? 1 : 0
+
+export type Or<A extends 1 | 0, B extends 1 | 0> = {
+  0: {
+    0: 0
+    1: 1
+  }
+  1: {
+    0: 1
+    1: 1
+  }
+}[A][B]
```

Comment:
[Link to Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcDyUA8BBOobAB2AJgM5wCMcAPnAAwA0cAQniASeVbXQHxwBeOAG8AUHHoAuEeIlT6siRWkVZAX1nKZc+ap1a9cDWoDa2ALonm50aKQo4AJWCkArgBt4QjJkYVeQA) for reviewers.

Reply:
It's not used atm but decided to leave it anyways, as it can be a useful type anyways.